IceCream/Classes/CreamAsset.swift

```diff
@@ -74,51 +67,79 @@ public class CreamAsset: Object {
     ///   - data: The file data
-    ///   - shouldOverwrite: Whether to try and save the file even if an existing file exists for the same object ID.
+    ///   - shouldOverwrite: Whether to try and save the file even if an existing file exists for the same object.
     /// - Returns: A CreamAsset if it was successful
-    public static func create(objectID: String, propName: String, data: Data, shouldOverwrite: Bool = true) -> CreamAsset? {
-        return CreamAsset(objectID: objectID,
-                          propName: propName,
-                          data: data,
-                          shouldOverwrite: shouldOverwrite)
+    public static func create(object: CKRecordConvertible, propName: String, data: Data, shouldOverwrite: Bool = true) -> CreamAsset? {
+        return create(objectID: object.recordID.recordName,
+                      propName: propName,
+                      data: data,
+                      shouldOverwrite: shouldOverwrite)
     }
 
     /// Creates a new CreamAsset for the given object with a URL
     ///
     /// - Parameters:
     ///   - object: The object the asset will live on
     ///   - propName: The unique property name to identify this asset. e.g.: Dog Object may have multiple CreamAsset properties, so we need unique `propName`s to identify these.
-    ///   - url: The URL of the file to store. Any path extension on the file (e.g. "mov") will be maintained
+    ///   - url: The URL where the file located
     /// - Returns: A CreamAsset if it was successful
     public static func create(object: CKRecordConvertible, propName: String, url: URL) -> CreamAsset? {
-
-        return CreamAsset(objectID: object.recordID.recordName,
-                          propName: propName,
-                          url: url)
+        return create(objectID: object.recordID.recordName,
+                      propName: propName,
+                      url: url)
+    }
+    
+    
+    /// Creates a new CreamAsset for the given objectID with a URL where asset locates
+    /// - Parameters:
+    ///   - objectID: The key to identify the object. Normally it's the recordName property of CKRecord.ID when recovering from CloudKit
+    ///   - propName: The unique property name to identify this asset. e.g.: Dog Object may have multiple CreamAsset properties, so we need unique `propName`s to identify these.
+    ///   - url: The location where asset locates
+    /// - Returns: The CreamAsset if creates successful
+    public static func create(objectID: String, propName: String, url: URL) -> CreamAsset? {
+        let creamAsset = CreamAsset(objectID: objectID, propName: propName)
+        if !FileManager.default.fileExists(atPath: creamAsset.filePath.path) {
+            do {
+                try FileManager.default.copyItem(at: url, to: creamAsset.filePath)
+            } catch {
+                /// Os.log copy item failed
+                return nil
+            }
+        }
+        return creamAsset
```

Comment:
Consider this case: If the url is the same but the binary data differs, the newly updated data will never be written to `creamAsset.filePath`. I think this is not what we expected?

Reply:
Personally, I was creating new entries in that case, but I guess with soft deletion it could make sense. Though that does bring back the importance of issue #107... ;)

Reply:
`shouldOverwrite` property already takes care of this case.